### PR TITLE
[clang][analyzer] Improve documentation of StreamChecker (NFC).

### PR DESCRIPTION
**Pull request by balazske**

### CODE REVIEW

File: clang/docs/analyzer/checkers.rst [hunk 1]
```diff
@@ -3020,44 +3020,82 @@ Check for misuses of stream APIs. Check for misuses of stream APIs: ``fopen, fcl
...
+  NULL is allowed.)
+* Use of stream after close.
+* Opened stream is not closed.
+* Read from a stream after end-of-file. (This is not a fatal error but reported
+  by the checker. Stream remains in EOF state and the read operation fails.)
+* Use of stream when the file position is indeterminate after a previous failed
+  operation. Some functions are allowed in this state.
+* Invalid 3rd ("``whence``") argument to ``fseek``.
+
+The checker is not capable of maintaining a relation between integer file
```
Comment (NagyDonat) [inline clang/docs/analyzer/checkers.rst:3040]:
```suggestion
The checker does not track the correspondence between integer file
```
and re-apply line wrapping to the paragraph (if needed)

**Review by NagyDonat — APPROVED** [review]
Looks reasonable.

**Review by benshi001 — APPROVED** [review]

File: clang/docs/analyzer/checkers.rst [hunk 2]
```diff
@@ -3020,44 +3020,82 @@ Check for misuses of stream APIs. Check for misuses of stream APIs: ``fopen, fcl
 
 alpha.unix.Stream (C)
 """""""""""""""""""""
-Check stream handling functions: ``fopen, tmpfile, fclose, fread, fwrite, fseek, ftell, rewind, fgetpos,``
-``fsetpos, clearerr, feof, ferror, fileno``.
+Check C stream handling functions:
+``fopen, fdopen, freopen, tmpfile, fclose, fread, fwrite, fgetc, fgets, fputc, fputs, fprintf, fscanf, ungetc, getdelim, getline, fseek, fseeko, ftell, ftello, fflush, rewind, fgetpos, fsetpos, clearerr, feof, ferror, fileno``.
+
+The checker maintains information about the C stream objects (``FILE *``) and
+can detect error conditions related to use of streams. The following conditions
```
[the remark this answers is not in the document]
Comment (balazske) [reply clang/docs/analyzer/checkers.rst:3027]: I do not know but "can detect error conditions related to use of streams" is good in itself so I would leave it.

File: clang/docs/analyzer/checkers.rst [hunk 3]
```diff
@@ -3020,44 +3020,82 @@ Check for misuses of stream APIs. Check for misuses of stream APIs: ``fopen, fcl
...
+are detected:
+
+* The ``FILE *`` pointer passed to the function is NULL. (At ``fflush``
+  NULL is allowed.)
+* Use of stream after close.
```
[the remark this answers is not in the document]
Comment (balazske) [reply clang/docs/analyzer/checkers.rst:3032]: This is not meant to be a function name, just "close of a stream" (I do not know why it is displayed in blue color).

File: clang/docs/analyzer/checkers.rst [hunk 4]
```diff
@@ -3020,44 +3020,82 @@ Check for misuses of stream APIs. Check for misuses of stream APIs: ``fopen, fcl
...
+The checker maintains information about the C stream objects (``FILE *``) and
+can detect error conditions related to use of streams. The following conditions
+are detected:
+
+* The ``FILE *`` pointer passed to the function is NULL (the single exception is
+  ``fflush`` where NULL is allowed).
+* Use of stream after close.
+* Opened stream is not closed.
+* Read from a stream after end-of-file. (This is not a fatal error but reported
+  by the checker. Stream remains in EOF state and the read operation fails.)
```
Comment (steakhal) [inline clang/docs/analyzer/checkers.rst:3035]: I can see `NULL` and `EOF` (maybe others) spelled in the docs without double backticks.
Do you think we should escape those to have verbatim code highlighting?

Comment (balazske) [reply clang/docs/analyzer/checkers.rst:3035]: I have the feeling that these words (NULL, EOF) are relatively commonly used (NULL is an abbrevation for null pointer, EOF = end-of-file). Probably `NULL` can be better as keyword (code format), but by EOF I did not mean the numeric constant, just "end of file".

**Pull request merged**